src/core/tests/hpc_io_looper.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Microsoft Corporation
+ *
+ * -=- Robust Distributed System Nucleus (rDSN) -=-
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/*
+ * Description:
+ *     Unit-test for io_looper.
+ *
+ * Revision history:
+ *     Nov., 2015, @xiaotz (Xiaotong Zhang), first version
+ *     xxxx-xx-xx, author, fix bug about xxx
+ */
+
+#include <fstream>
+#include <sstream>
+#include <vector>
+#include <string>
+#include <queue>
+#include <iostream>
+#include <sys/eventfd.h>
+#include <dsn/internal/aio_provider.h>
+#include <gtest/gtest.h>
+#include <dsn/service_api_cpp.h>
+#include <dsn/internal/priority_queue.h>
+#include "../core/group_address.h"
+#include "../core/command_manager.h"
+#include "../core/service_engine.h"
+#include "test_utils.h"
+#include "../tools/hpc/io_looper.h"
+
+
+# ifdef __linux__
+# include <libaio.h>
+# endif
+
+DEFINE_TASK_CODE_AIO(LPC_AIO_TEST, TASK_PRIORITY_COMMON, THREAD_POOL_TEST_SERVER);
+
+namespace dsn{
+namespace test{
+    class io_looper_for_test : public dsn::tools::io_looper
+    {
+    public:
+        io_looper_for_test() : io_looper(){}
+        virtual ~io_looper_for_test(){}
+    public:
+        virtual bool is_shared_timer_queue() override
+        {
+            if(io_looper::is_shared_timer_queue())
+                return false;
+            else
+                return true;
+        }
+    };
+}}
+
+TEST(tools_hpc, io_looper)
```

Comment:
fixed in https://github.com/qinzuoyan/rDSN/commit/1961ec28cab20607210a3afb26d927fb46a52b1a